src/frontend/src/optimizer/logical_optimization.rs

```diff
@@ -411,6 +417,8 @@ impl LogicalOptimizer {
 
         plan = plan.optimize_by_rules(&PROJECT_REMOVE);
 
+        plan = plan.optimize_by_rules(&PULL_UP_HOP);
```

Comment:
Should we apply this rule to streaming queries ?

Reply:
Remove it later seems streaming queries will not benefit from it.🥵 So seems we also shouldn't apply PROJECT_JOIN_TRANSPOSE in streaming queries?